springboot-jersey-app/delivery/src/main/java/com/wfsample/delivery/DeliveryController.java

```diff
@@ -0,0 +1,41 @@
+package com.wfsample.delivery;
+
+import com.wfsample.common.dto.DeliveryStatusDTO;
+import com.wfsample.common.dto.PackedShirtsDTO;
+import com.wfsample.service.DeliveryApi;
+
+import java.util.UUID;
+
+import javax.ws.rs.core.Response;
+
+/**
+ * Controller for delivery service which is responsible for dispatching shirts returning tracking
+ * number for a given order.
+ *
+ * @author Hao Song (songhao@vmware.com).
+ */
+public class DeliveryController implements DeliveryApi {
+
+  @Override
+  public Response dispatch(String orderNum, PackedShirtsDTO packedShirts) {
+    if (orderNum.isEmpty()) {
+      return Response.status(Response.Status.BAD_REQUEST).entity("Invalid Order Num").build();
+    }
+    if (packedShirts == null || packedShirts.getShirts() == null ||
+        packedShirts.getShirts().size() == 0) {
+      return Response.status(Response.Status.BAD_REQUEST).entity("no shirts to deliver").build();
+    }
+    String trackingNum = UUID.randomUUID().toString();
+    System.out.println("Tracking number of Order:" + orderNum + " is " + trackingNum);
+    return Response.ok(new DeliveryStatusDTO(orderNum, trackingNum,
```

Comment:
What we should also do here is add the dispatch items to a Queue.
And have scheduled executor service that dequeues from that queue and delivers the order.
So 2 TODOs that we can add -
1) Add a gauge to monitor the size of that queue.
2) Increment a counter when delivered.